ee/enc/vault_ee.go

```diff
@@ -0,0 +1,147 @@
+// +build !oss
+
+/*
+ * Copyright 2020 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package enc
+
+import (
+	"io/ioutil"
+
+	"github.com/dgraph-io/dgraph/x"
+	"github.com/hashicorp/vault/api"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// Configuration options of Vault.
+const (
+	vaultAddr         = "vault_addr"
+	vaultRoleIDFile   = "vault_roleid_file"
+	vaultSecretIDFile = "vault_secretid_file"
+	vaultPath         = "vault_path"
+	vaultField        = "vault_field"
+)
+
+// RegisterVaultFlags registers the required flags to integrate with Vault.
+func registerVaultFlags(flag *pflag.FlagSet) {
+	// The following are Vault options. Applicable for alpha, live, bulk, debug, restore sub-cmds
+	flag.String(vaultAddr, "http://localhost:8200",
+		"Vault server's address in the form http://ip:port.")
+	flag.String(vaultRoleIDFile, "",
+		"File containing Vault role-id used for approle auth.")
+	flag.String(vaultSecretIDFile, "",
+		"File containing Vault secret-id used for approle auth.")
+	flag.String(vaultPath, "dgraph",
+		"Vault kv store path.")
+	flag.String(vaultField, "enc_key",
+		"Vault kv store field whose value is the encryption key.")
+}
+
+// vaultKeyReader implements the KeyReader interface. It reads the key from vault server.
+type vaultKeyReader struct {
+	addr     string
+	roleID   string
+	secretID string
+	path     string
+	field    string
+}
+
+func newVaultKeyReader(cfg *viper.Viper) (*vaultKeyReader, error) {
+	v := &vaultKeyReader{
+		addr:     cfg.GetString(vaultAddr),
+		roleID:   cfg.GetString(vaultRoleIDFile),
+		secretID: cfg.GetString(vaultSecretIDFile),
+		path:     cfg.GetString(vaultPath),
+		field:    cfg.GetString(vaultField),
+	}
+
+	if v.addr == "" || v.path == "" || v.field == "" {
+		return nil, errors.Errorf("%v, %v or %v is missing", vaultAddr, vaultPath, vaultField)
+	}
+
+	if v.roleID != "" && v.secretID != "" {
+		return v, nil
+	}
+	return nil, errors.Errorf("%v and %v must both be specified",
```

Comment:
I prefer writing the error conditions in the `if`. It makes it easier to read code. I would've written this as
```go
if rolID == "" || secret == ""  { return err .. }
```